File: src/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3 
#這個檔案是 Python 3 程式，Linux 執行時請自動找 Python 3 來跑

import rclpy # import the ROS2 Python library
from rclpy.node import Node #  import the Node class from the ROS2 Python library


class Mynode(Node):
    def __init__(self):
        super().__init__("py_test")
        self.counter_ = 0 
        self.get_logger().info("Hellow worldddd") # print "Hello world" in the terminal
        self.create_timer(1.0, self.timer_callback) # create a timer, call the function "timer_callback" every 1 second

    def timer_callback(self):
        self.get_logger().info("YOOOO" + str(self.counter_))
        self.counter_ += 1
def main(args=None): # create input var "args" which intial value is "none"
    rclpy.init(args=args) # initialize the ROS2 communication
    node = Mynode()  # create a node object
    rclpy.spin(node)    # do not exit the program, keep it running until user press Ctrl+C
    rclpy.shutdown() # shutdown the ROS2 communication

if __name__ == "__main__":  #if this file is the main program, then run the main function
    main()

File: src/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp" // Include the ROS 2 C++ client library

class MyNode : public rclcpp::Node // Define a new class that inherits from rclcpp::Node
{
    public:
        MyNode() : Node("cpp_test"), counter_(0) //  Constructor that initializes the node with the name "cpp_test"
        {
            RCLCPP_INFO(this->get_logger(), "Hello world"); // Log a message when the node is created
            timer_ = this->create_wall_timer(std::chrono::seconds(1), std::bind(&MyNode::timer_call_back, this)); // Create a timer that calls the timer_call_back function every second
        }
    private:
        void timer_call_back()
        {
                RCLCPP_INFO(this->get_logger(), "YOOOO %d", counter_);
                counter_++;
        }
        rclcpp::TimerBase::SharedPtr timer_; // A shared pointer to a timer object
        int counter_;   //  A counter variable to keep track of the number of times the timer callback has been called
};


int main(int argc, char **argv) // The main function is the entry point of the program
{
    rclcpp::init(argc, argv); // Initialize the ROS 2 client library
    auto node = std::make_shared<MyNode>(); // Create a new node
    rclcpp::spin(node); // Keep the node alive until it is shut down
    rclcpp::shutdown(); // Shutdown the ROS 2 client library

    return 0; // Return success
}
